### DISCUSSION TITLE
Differentiate requeued object updates from normal reconcilations

### DISCUSSION THREAD

**Discussion by yalctay93** [kubernetes-sigs/kubebuilder#2247] [Support] [answered]
Hello,

we are requeuing our objects so that they are reconciled in a time interval, e.g:

```
	d := 300 * time.Second
	return ctrl.Result{RequeueAfter: d}, nil

We are using the EventFilter predicate.GenerationChangedPredicate

func (reconcile *ObjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
	return ctrl.NewControllerManagedBy(mgr).
		For(&object{}).
		WithEventFilter(predicate.GenerationChangedPredicate{}).
		Owns(&object{}).
		Complete(reconcile)
}

```

All of the above works flawlessly, the objects are requeued after 300 seconds however, they also run through our entire reconcile code and thus perform client.Update() on our subresources. Usually, in k8s when you use the ```kubectl apply``` command, k8s will report "unchanged" if you've not changed anything.

Is there a way to figure out if the object is "unchanged" by k8s when using client.Update() function? The object is only returning an error, or is there any way for me to figure out if the route object that is being reconciled is a requeued object?

Background is, that I do not want to log "subresource has been updated" each time if the reconcile is just a requeued object. If I were able to see if the object was reconciled by a requeue, or if the client.Update() returned an "unchanged" I'd be able to not log "subresource has been updated". I just want to differentiate updates triggered by a requeued object, and a real reconcile that is triggered by someone applying our crd.

Thanks!

**Comment by camilamacedo86** [comment, marked as answer]
Hello,

For the use case you've described, the most efficient approach would be to leverage Kubernetes' capability to watch resources. This way, instead of reconciling at fixed intervals, you'll only reconcile when there's an actual change to the observed resource.

By watching the resources, you can ensure that your reconcile loop is triggered only when there are genuine updates to the resources. This approach eliminates the issue of having to differentiate between a requeued reconcile and a real one, as you'll only be reacting to actual changes.

For more details on how to set this up, refer to: [Watching Resources](https://book.kubebuilder.io/reference/watching-resources).

I hope this addresses your concern. Let me know if you have any further questions.

Best regards.

Closing this one as answered feel free to re-open if you see that is required
